### ISSUE TITLE
[RESOLVED] Race won't start on Windows

### ISSUE THREAD

**Issue by jonfen** [cwhitney/GoldsprintsFx#37]
TL;DR: Update the Arduino drivers for windows to the ones provided by Arduino, the default drivers provided by windows will pass the serial connection test (https://www.opensprints.com/howto_test_electronics.php) but they DO NOT work with SilverSprints.  Nothing will happen when you press Start.

There are a couple of ways to accomplish this:

1.  Install the Arduino IDE (https://www.arduino.cc/en/software) and say "Install" to **ALL** Windows Security alerts that are asking "Would you like to install this device software?"

![image](https://user-images.githubusercontent.com/56961/179018371-8ba0c698-54fe-4b2b-885a-7c429942656f.png)
![image](https://user-images.githubusercontent.com/56961/179018790-009228cd-c7a4-48f9-a909-cace3f85a2b4.png)

2. Install the Arduino IDE (https://www.arduino.cc/en/software), open up Windows "Device Manager", right-click on the COM port, and select "Update driver" pointing it to "C:\Program Files (x86)\Arduino\drivers", select Arduino Uno.

One way to verify that this is your issue is by going to Windows Event Viewer to see what the errors for SilverSprints are.  In this case, it says "The semaphore is set and cannot be closed"

![image](https://user-images.githubusercontent.com/56961/179017591-a4fde9af-513e-40b5-8258-34bd7f5031e0.png)

**Comment by jonfen** [comment]
@cwhitney given that this isn't clear to most users, is there a way to programmatically check on windows to see if the correct drivers are installed and provide instruction?  or better yet, try to install the right ones?

**Comment by 2pick** [comment]
At last 2 pair gold sprint roller’s I use arduino uno on 16u2. Stability connect & work with win7, 10 & mac.

Some problem was with arduino on ch340 chip & ftdi without installed driver.

**Comment by jonfen** [comment]
> At last 2 pair gold sprint roller’s I use arduino uno on 16u2. Stability connect & work with win7, 10 & mac.
> 
> Some problem was with arduino on ch340 chip & ftdi without installed driver.

Software isn't the only thing that needs updating.  While @dhalbert could probably explain in more depth, OpenSprints simply 
 recommends updating the hardware to an **Arduino Uno Rev3** as an easy and inexpensive solution for refreshing older OpenSprints systems.  Instructions: https://www.opensprints.com/howto_upgrade_opensprints_hub_arduino.php

**Comment by pcwatkins** [comment]
I can confirm the change of drivers has now made it work on a Surface Pro!
